### PR TITLE
update to CSS 7 and switch to vitest

### PR DESCRIPTION
**Pull request by elf-pavlik**
There is one test failing, and I'm going to update this PR later today.

CSS 7 uses the newer `oidc-provider`, which is ESM. Jest was just refusing to cooperate, and https://vitest.dev simply worked out of the box with zero configuration.

### CODE REVIEW

**Comment by elf-pavlik** [comment]
okie dokie, the tests are passing https://github.com/elf-pavlik/solid-notification-client/tree/css7

Once merged, it will also start running tests using GitHub action.

**Comment by woutslabbinck** [comment]
TL;DR: 
I can not build the code on branch `css7` neither with npm, nor with yarn.
Could you give some more information @elf-pavlik or help me out?

Some information about my machine

node version: v18.18.2
npm version: 9.8.1
yarn version: 4.1.0 (comes later in the story)

What I did:
```sh 
git clone git@github.com:elf-pavlik/solid-notification-client.git
cd solid-notification-client/
git checkout css7 
npm install
npm run build
```

Which resulted in 
```sh
> build
> lerna run build

lerna notice cli v8.1.5

 Lerna (powered by Nx)   Running target build for 4 projects

   ✖  solid-test-utils:build
      > solid-test-utils@0.1.2 build
      > rollup --config rollup.config.mjs
      
      [!] SyntaxError: Unexpected token 'with'
          at ESMLoader.moduleStrategy (node:internal/modules/esm/translators:119:18)
          at ESMLoader.moduleProvider (node:internal/modules/esm/loader:468:14)
          at async link (node:internal/modules/esm/module_job:68:21)
      
      
      npm ERR! Lifecycle script `build` failed with error: 
      npm ERR! Error: command failed 
      npm ERR!   in workspace: solid-test-utils@0.1.2 
      
   ✖  @solid-notifications/types:build
      > @solid-notifications/types@0.1.2 build
      > rollup --config rollup.config.mjs
      
      [!] SyntaxError: Unexpected token 'with'
          at ESMLoader.moduleStrategy (node:internal/modules/esm/translators:119:18)
          at ESMLoader.moduleProvider (node:internal/modules/esm/loader:468:14)
          at async link (node:internal/modules/esm/module_job:68:21)
      
      
      npm ERR! Lifecycle script `build` failed with error: 
      npm ERR! Error: command failed 
      npm ERR!   in workspace: @solid-notifications/types@0.1.2 

 Lerna (powered by Nx)   Ran target build for 4 projects (403ms)
   ✔  0/2 succeeded [0 read from cache]

   ✖  2/2 targets failed, including the following:

      - solid-test-utils:build
      - @solid-notifications/types:build
```

I thought at that moment that I maybe needed yarn ( version 4.1.0) so I tried that way, but the errors did not make sense to me (something about requiring either `string-width` by `lerna@npm:8.1.5`. I think this a my machine problem).

What I did:
```sh 
cd ../
rm -rf solid-notification-client/
git clone git@github.com:elf-pavlik/solid-notification-client.git
cd solid-notification-client/
git checkout css7 
yarn install
yarn build
```

Which resulted in 
```sh
yarn build
/home/wouts/Documents/repos/Solid/solid-notification-client/.pnp.cjs:19066
      Error.captureStackTrace(firstError);
            ^

Error: lerna tried to access string-width, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: string-width
Required by: lerna@npm:8.1.5
```

Currently, I cannot do much to test whether it works or not apart from saying that the changes do not work on my machine.
I hope someone can point me to how I should run it so I can test the changes :)

**Comment by elf-pavlik** [comment]
Thanks @woutslabbinck !

The issue is with node 18. `with` is used to import JSON in rollup config used to build hybrid modules. 

Node 18 is already in maintenance mode with eol in 10 months. Tests on CI run with current LTS 20 and current 22.

Do we care to have it work in node 18. If  we do is it enough to have build npm package to work or the build step as well?

**Comment by woutslabbinck** [comment]
That makes sense, by using node v20 it does work. But I would then maybe add that to the README under [Development Setup](https://github.com/o-development/solid-notification-client/blob/main/README.md?plain=1#L57)

```sh
git clone git@github.com:elf-pavlik/solid-notification-client.git
cd solid-notification-client/
git checkout css7 
nvm use
npm install
npm run build
```

Worked. I tested with a CSS v7 by running `npx @solid/community-server -c @css:config/file-root.json -f data/` in the `solid-notification-client` directory and then executing the readme code to retrieve a websocketchannel :)

javascript code run:
```js
const { SubscriptionClient} = require('./packages/subscription')
const { ChannelType} = require('./packages/types')

async function main(){
    const topic = "http://localhost:3000/";
    const channelType = ChannelType.WebSocketChannel2023;
    
    const client = new SubscriptionClient(fetch);
    const notificationChannel = await client.subscribe(topic, channelType);
    
    console.log(notificationChannel);
}
main()
```
and the result

```json
{
  id: 'http://localhost:3000/.notifications/WebSocketChannel2023/452cb4ca-40ee-4c5b-9452-cbaad84bc400',
  type: 'http://www.w3.org/ns/solid/notifications#WebSocketChannel2023',
  topic: 'http://localhost:3000/',
  receiveFrom: 'ws://localhost:3000/.notifications/WebSocketChannel2023/452cb4ca-40ee-4c5b-9452-cbaad84bc400'
}
```

**Comment by elf-pavlik** [comment]
Thanks. I assume that `npm test` also works, as it passes in CI workflow this PR is also adding
https://github.com/elf-pavlik/solid-notification-client/actions/runs/9672002221

**EDIT**: I updated the README, please approve the PR if you think it is ok to merge.

**Review by woutslabbinck — APPROVED** [review]
I also just ran the test and that works locally with node 20.
Since, the node version is documented in the README and everything works, it looks good for me.
